phone/src/apps/settings/components/WallpaperModal.tsx

```diff
@@ -21,59 +21,43 @@ export default function WallpaperModal() {
   const { customWallpaperModal, setCustomWallpaperModal } = useCustomWallpaperModal();
   const [settings, setSettings] = useSettings();
   const { t } = useTranslation();
-  const [value, setValue] = useState(settings.customWallpaper ? settings.customWallpaper : '');
-  const classes = useStyles();
+  const [value, setValue] = useState('');
+  const { addAlert } = useSnackbar();
 
-  const isImage = (url) => {
+  const isImageAndUrl = (url) => {
     return /(http(s?):)([/|.|\w|\s|-])*\.(?:jpg|png|jpeg|gif)/g.test(url);
   };
 
-  const handleCloseModal = () => {
-    setCustomWallpaperModal(false);
-  };
-
   const handleSettingChange = (key: string | number, value: unknown) => {
     setSettings({ ...settings, [key]: value });
   };
 
   const handleNewWallpaper = () => {
-    handleSettingChange('wallpaper', { label: 'Custom Wallpaper', value });
-    setCustomWallpaperModal(false);
+    if (isImageAndUrl(value)) {
+      handleSettingChange('wallpaper', {
+        label: t('APPS_SETTINGS_OPTIONS_CUSTOM_WALLPAPER'),
```

Comment:
If we set just the translation key here rather than the actual translation, we can just directly translate it on both ends rather than storing a hardcoded locale.

Reply:
Yep, was thinking about that.

Reply:
Still needs changing